### PR TITLE
More robust hdf5, along with more refined I/O from cmake

### PR DESCRIPTION
**Pull request by mauneyc-LANL**
## PR Summary

Addressing issues @jdolence encountered with HDF5 resolution. Some small rearranging of dependency logic, and a less chatty cmake (by default, can be adjusted)

I had attempted to use `module` mode to resolve `find_package(HDF5)` with a distributed `FindHDF5.cmake`, but I think I made a mess of it and Josh hit an issue that arose from that. The large change here is to revert to a more "standard" approach that relies on the implementation of "official" Kitware CMake module (https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/FindHDF5.cmake).

For reference, it can be informative to expose some trace information on `find_package` when configuring

```bash
cmake --debug-find-pkg=HDF5 .. -DSINGULARITY_USE_HDF5=ON #...
```

Other notable changes:
- Output has been reduced in the default log level. A more chatty output can be restored using `--log-level=VERBOSE` option: `cmake --log-level=VERBOSE .. #...`
- some cleanup of `singularity_import_*` macros

@jonahm-LANL @dholladay00 @jhp-lanl 

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -208,7 +208,7 @@ if(SINGULARITY_USE_KOKKOS)
   # type: soft submodule 
   singularity_import_dependency(
     PKG Kokkos
-    TARGET Kokkos::kokkos
+    TARGETS Kokkos::kokkos
```
Comment (Yurlungur) [inline CMakeLists.txt:211]: what is the difference between these?

**Review by Yurlungur — COMMENTED** [review]
Don't forget to update changelog

File: CMakeLists.txt (hunk 1 above)
Comment (mauneyc-LANL) [reply CMakeLists.txt:211]: An `S`

the macro took as input `TARGET` and `TARGETS`, These did the same thing, but just bifurcated into singular and plural. I changed this to only one input `TARGETS`

**Review by Yurlungur — APPROVED** [review]
Looks good to me. @jdolence can you double check this resolves the issues you were seeing?

**Comment by dholladay00** [comment]
@jdolence what is your timeline on testing this? I'd like to get this in soon.

**Comment by dholladay00** [comment]
@mauneyc-LANL can you resolve all incomplete items in the checklist? When completed, I'll merge. 

I suspect downstream testing will be necessary.

**Comment by Yurlungur** [comment]
Yeah I think @jdolence is not going to test this. Assuming tests pass I'm fine with merging without waiting for him.

File: CMakeLists.txt [hunk 2]
```diff
@@ -285,22 +285,12 @@ if (SINGULARITY_USE_HDF5)
   enable_language(C)
```
Comment (junghans) [inline CMakeLists.txt:285]: @mauneyc-LANL We can drop this for cmake>=3.26, see: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8015

Comment (mauneyc-LANL) [reply CMakeLists.txt:285]: yes; Indeed, we don't require to do this for `spiner`, and I'm working on a PR for singularity to follow that approach.

**Pull request merged**